src/main/java/bg/codexio/springframework/boot/adapter/GraphQLHttpFilterAdapter.java

```diff
@@ -0,0 +1,759 @@
+                    query,
+                    variables
+            );
+        } catch (Exception e) {
+            this.logger.error(
+                    e.getMessage(),
+                    e
+            );
+
+            return new FilterRequestWrapper<>();
+        }
+    }
+
+    /**
+     * Processes a POST request, extracting filter data from the request body.
+     *
+     * <p>
+     * This method parses the request body into a JSON map to retrieve the
+     * "query" and "variables" fields. It then prepares and converts these into
+     * a {@link FilterRequestWrapper} by calling {@code prepareAndConvertQuery}.
+     * </p>
+     *
+     * @param request the {@link HttpServletRequest} containing filter data
+     *                in the request body
+     * @param <T>     the type of the result in the {@link FilterRequestWrapper}
+     * @return a {@link FilterRequestWrapper} representing the filter,
+     * or an empty wrapper if an error occurs
+     */
+    private <T> FilterRequestWrapper<T> processPostRequest(HttpServletRequest request) {
+        try {
+            var jsonMap = createJsonMap(request);
+
+            var query = (String) jsonMap.get("query");
+            var variables = (String) jsonMap.get("variables");
+
+            return this.prepareAndConvertQuery(
+                    query,
+                    variables
+            );
+
+        } catch (Exception e) {
+            this.logger.error(
+                    e.getMessage(),
+                    e
+            );
+            return new FilterRequestWrapper<>();
+        }
+    }
+
+    /**
+     * Prepares and converts a GraphQL query into a
+     * {@link FilterRequestWrapper}.
+     *
+     * <p>
+     * This method substitutes variables into the query if provided, extracts
+     * the filter data from the query, and adapts it into a
+     * {@link FilterRequestWrapper}.
+     * If the query contains complex filters, it uses {@code
+     * graphQLComplexFilterAdapter}
+     * to handle the adaptation.
+     * </p>
+     *
+     * @param query     the GraphQL query string to process
+     * @param variables the JSON string of variables to substitute into the
+     *                  query
+     * @param <T>       the type of the result in the
+     *                  {@link FilterRequestWrapper}
+     * @return a {@link FilterRequestWrapper} containing the adapted filter,
+     * or an empty wrapper if an error occurs or if no filter is found
+     */
+    private <T> FilterRequestWrapper<T> prepareAndConvertQuery(
+            String query,
+            String variables
+    ) {
+        try {
+            if (variables != null) {
+                query = substituteVariablesInQuery(
+                        query,
+                        createVariablesMap(variables)
+                );
+            }
+
+            var extractedFilter = extractFilterBody(query);
+            if (extractedFilter.isPresent()) {
```

Comment:
Shouldn't this be a `map()` and the below one `orElseGet`?

Reply:
done